beacon_node/operation_pool/src/sync_aggregate_id.rs

```diff
@@ -0,0 +1,43 @@
+use serde_derive::{Deserialize, Serialize};
+use ssz::ssz_encode;
+use ssz_derive::{Decode, Encode};
+use types::sync_committee_contribution::SyncAggregateData;
+use types::{ChainSpec, Domain, Epoch, EthSpec, Fork, Hash256, Slot};
+
+/// Serialized `SyncAggregateData` augmented with a domain to encode the fork info.
+#[derive(
+    PartialEq, Eq, Clone, Hash, Debug, PartialOrd, Ord, Encode, Decode, Serialize, Deserialize,
+)]
+pub struct SyncAggregateId {
+    v: Vec<u8>,
```

Comment:
I'd be tempted to go with a fixed-length object here if possible. If not, perhaps a `SmallVec` would be good, but not a big deal.

Fixed length would be ideal, IMO. Helps ensure correctness. I haven't looked into if this is possible though.

Reply:
Seems like the `Vec<u8>` is useful with attestations to compare domains while iterating. But since we only have one sync aggregate to check each slot, I don't think it's necessary. So I updated it to:

```
pub struct SyncAggregateId {
    pub slot: Slot,
    pub beacon_block_root: Hash256,
    pub domain: Hash256,
}
```